lib/accesscontrol/permitted_action_query.rb

```diff
@@ -31,6 +31,28 @@ def can?(action_id, object_type)
       end
     end
 
+    # Allow permission on a general action in the given namespace represented by object_type.
+    # A grant is universally unique and is enforced at the database level.
+    #
+    # @param action_id [Integer] The action to grant for the object
+    # @param object_type [String] The namespace of the given action_id.
+    # @return [nil] Returns nil if successful, otherwise will raise an Error (AccessControl::CouldNotGrantError).
+    #
+    # @example
+    #   # Allow the user access to posts
+    #   AccessControl::Query.new(user).grant(3, "posts")
+    def grant(action_id, object_type)
+      PermittedAction.create!(
+        id: SecureRandom.uuid,
+        actor: @actor,
+        action: action_id,
+        object_type: String(object_type)
+      )
+      nil
+    rescue => e
+      raise AccessControl::CouldNotGrantError.new("Could not grant action #{action_id} on object #{object_type} for actor #{@actor}")
```

Comment:
Like I asked above, should we silently `rescue` from `PG::UniqueViolation`?

Reply:
see question above